pkg/curatedpackages/packageinstaller.go

```diff
@@ -10,6 +10,7 @@ import (
 
 type PackageController interface {
 	InstallController(ctx context.Context) error
+	ValidateControllerDoesNotExist(ctx context.Context) error
```

Comment:
You could abbreviate this to `IsInstalled` and then use `!controller.IsInstalled(ctx)`. Typically these functions are better defined as affirming something _is_ rather than _is not_. That's because its easier to understand `!Isinstalled()` and `IsInstalled()` as opposed to `IsNotInstalled()` and `!IsNotInstalled()` (the latter being the confusing part because its a double negative).

Reply:
Agreed with @chrisdoherty4 
However, where is this being used?

Reply:
It is currently used here: https://github.com/aws/eks-anywhere/blob/72414be6c5bfe3caccaf4cbf8fc2de157c2a2c82/cmd/eksctl-anywhere/cmd/installpackagecontroller.go#L92.

Reply:
Ahh. I think we should probably change the factory so it stores a concrete implementation as opposed to the interface.

Reply:
What would the benefit of storing the concrete implementation rather than the interface?

Reply:
The factory is not using this entity, is only building it. And it's building a concrete type directly. In general, is better to return concrete implementations and accept interfaces.

One pretty straight forward benefit: we had to add/change this method in an interface defined in the `curatedpackages` package when the (I think?) only receiver is the cmd package. Which, in fact, doesn't care about receiving interfaces or not because is not even being tested and already depends on the `curatedpackages`. If the factory was returning a concrete type, we wouldn't need to update that interface.

IMO interfaces in the deps struct make sense only when we conditionally build different implementations depending on some external configuration (like we do with providers).